### ISSUE TITLE
Mock PE: Level 2

### ISSUE THREAD

**Issue by chooweixiang98** [nus-cs2030/2021-s2#49]
![image](https://user-images.githubusercontent.com/26595217/109090860-ae476100-774e-11eb-8daf-57e8c6dfeff0.png)

For these 3 lines: 
1   Employee employee = new Employee(sc.next(), sc.next(), sc.nextDouble(), adjustment);
2   employee.setSalaryIncrease(sc.nextDouble());
3   System.out.println(employee);

In line 2, the setSalaryIncrease( ) is called from the employee object and there's no other Employee class variable taking in the return value e.g., Employee emp = employee.setSalaryIncrease(sc.nextDouble()); 

Does that mean that we need to make the annualRaiseRate variable in the Employee class mutable so that the setter method can change the annualRaiseRate? 
![image](https://user-images.githubusercontent.com/26595217/109091445-c66bb000-774f-11eb-96db-21f191fece3d.png)
![image](https://user-images.githubusercontent.com/26595217/109091484-d71c2600-774f-11eb-815f-066e77e68acd.png)

I'm asking this because how Main.java was defined. 

My main concern here is immutability, but I'm not sure how to get around the codes defined in Main.java. Anyone knows the correct way to get around this without changing Main.java?

**Comment by raymondtanujaya** [comment]
Since no new object is return, I also think that salary increase should be mutable

**Comment by gableejh** [comment]
Yes in this case it has to be mutable in order to set the annual raise value

**Comment by Keithlimhs** [comment]
Yes i set my annualRaiseRate as mutable otherwise the setter method will not work. To adhere to immutability, our setter methods should return a new instance of the Employee class, however, the constructor of the Employee class does not take in annualRaiseRate as a parameter in the first place hence there is no point returning a new instance of Employee. I submitted my code onto CodeCrunch and it worked so i guess in this case it is allowed to be mutable. Hope this clarifies your doubts

**Comment by etbf** [comment]
yes i was also stuck at this line and from reading line 2, we can actually see that the fourth value is not taken in via the constructor but via a method, thus we have to set it to a value that we declare in the first place via a void method.

**Comment by Xeph-o** [comment]
Yes, annualRaiseRate should be mutable so that it can be changed.

**Comment by Radix-huang** [comment]
At the beginning stage, I was also concerned about immutability and made it a final one but after that, it did not make sense for the salary increase. So I think it should be mutable in this scenario.
